cache/git.go

```diff
@@ -20,10 +23,27 @@ func (s *gitSource) OpenLocal(c *Cache, checksum string) (*os.File, error) {
 	return f, errors.WithStack(err)
 }
 
+// applyGitHubToken modifies GitHub HTTPS URLs to include the token if available
+func applyGitHubToken(url string, logger ui.Logger) string {
+	if strings.HasPrefix(url, "https://github.com/") && os.Getenv(GitHubTokenEnv) != "" {
```

Comment:
I don't really mind the idea, but the token value should be plumbed through from [here](https://github.com/cashapp/hermit/blob/483aef37f656a2b6b1ad842b38e3eae695bb87f9/app/main.go#L175).

Reply:
You'll also likely want to adhere to the Hermit config that restricts which repos use authentication.